### PR TITLE
Initial Database Project with Change Tracking

### PR DESCRIPTION
**Pull request by keithnicholson**
Adds a database project to the samples folder.  It generates a database with Change Tracking enabled, including tracking set on the tables

### CODE REVIEW

**Comment by keithnicholson** [comment]
@microsoft-github-policy-service agree



Peace,

Keith Nicholson

________________________________

**Comment by Charles-Gagnon** [comment]
@dzsquared @kisantia @Benjin to help look over the sqlproj

But what would probably be best is to make this an [SDK-style sqlproj](https://learn.microsoft.com/en-us/sql/azure-data-studio/extensions/sql-database-project-extension-sdk-style-projects?view=sql-server-ver16) - that is the newer format that will reduce a lot of the clutter the old style has. Let us know if you have any questions and we'd be glad to help!

**Comment by keithnicholson** [comment]
@Charles-Gagnon

I had not looked into this yet.  I'll give it a try and push an update.

**Comment by dzsquared** [comment]
@keithnicholson - to be clear, the SDK-style projects aren't supported in Visual Studio yet so you may want to hold off if you're really attached to using Visual Studio instead of VS Code or Azure Data Studio for the database work.  Otherwise, SDK-style SQL projects are pretty nifty and much easier for use with dotnet etc.


https://github.com/microsoft/DacFx/blob/main/src/Microsoft.Build.Sql/docs/Converting-Existing.md


Your sqlproj file would condense down to:
```xml
<?xml version="1.0" encoding="utf-8"?>
<Project DefaultTargets="Build" ToolsVersion="4.0">
  <Sdk Name="Microsoft.Build.Sql" Version="0.1.12-preview" />
  <PropertyGroup>
    <Name>DatabaseProjectAFSqlExtensions</Name>
    <SchemaVersion>2.0</SchemaVersion>
    <ProjectVersion>4.1</ProjectVersion>
    <ProjectGuid>{34897831-8d8d-4ca6-b1c2-8ea1535b4c05}</ProjectGuid>
    <DSP>Microsoft.Data.Tools.Schema.Sql.Sql160DatabaseSchemaProvider</DSP>
    <DefaultFileStructure>BySchemaAndSchemaType</DefaultFileStructure>
    <IsChangeTrackingOn>True</IsChangeTrackingOn>
    <ChangeTrackingRetentionPeriod>7</ChangeTrackingRetentionPeriod>
  </PropertyGroup>
</Project>
```

File: samples/DatabaseProjectAFSqlExtensions/DatabaseProjectAFSqlExtensions.sqlproj [hunk 1]
```diff
@@ -0,0 +1,86 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Benjin) [inline samples/DatabaseProjectAFSqlExtensions/DatabaseProjectAFSqlExtensions.sqlproj:1]: Have you considered using the newer and simpler SDK-style .sqlproj format supported in Azure Data Studio?  It'd really clean this file up!

Comment (keithnicholson) [reply samples/DatabaseProjectAFSqlExtensions/DatabaseProjectAFSqlExtensions.sqlproj:1]: Others have mentioned it.  I had not seen it as an option until today.   I'll be doing it tomorrow.

**Comment by keithnicholson** [comment]
I have created a [SDK-style sqlproj](https://learn.microsoft.com/en-us/sql/azure-data-studio/extensions/sql-database-project-extension-sdk-style-projects?view=sql-server-ver16) locally, yet I have trouble building it.  I found an issue on the project that I have added my comments at :  https://github.com/microsoft/DacFx/issues/330.

I'll update my PR later.

**Comment by chlafreniere** [comment]
> I have created a [SDK-style sqlproj](https://learn.microsoft.com/en-us/sql/azure-data-studio/extensions/sql-database-project-extension-sdk-style-projects?view=sql-server-ver16) locally, yet I have trouble building it. I found an issue on the project that I have added my comments at : [microsoft/DacFx#330](https://github.com/microsoft/DacFx/issues/330).
> 
> I'll update my PR later.

Thanks @keithnicholson! I've worked with the owner of the sql build sdk, and he's taking a look.

File: samples/DatabaseProjectAFSqlExtensions/DatabaseProjectAFSqlExtensions.sqlproj (hunk 1 above)
Comment (Charles-Gagnon) [inline samples/DatabaseProjectAFSqlExtensions/DatabaseProjectAFSqlExtensions.sqlproj:1]: I'd suggest just putting this in https://github.com/Azure/azure-functions-sql-extension/tree/main/samples/Database and using the scripts there. That way we don't have to remember to go update these ones as well when we add/modify anything.

Comment (keithnicholson) [reply samples/DatabaseProjectAFSqlExtensions/DatabaseProjectAFSqlExtensions.sqlproj:1]: When I update the PR with the SDK-style, you will see that I named the folder differently. When it gets committed, I think I will suggest removing the prior sql scripts.

**Comment by Charles-Gagnon** [comment]
We're going to close this for now since there hasn't been any updates in a while, but feel free to reopen if you get a chance to work on this some more!

**Pull request closed without merging**